validator/validator.go

```diff
@@ -99,62 +99,114 @@ func (v *Validator) ValidateToken(ctx context.Context, tokenString string) (inte
 		return nil, fmt.Errorf("could not parse the token: %w", err)
 	}
 
-	if string(v.signatureAlgorithm) != token.Headers[0].Algorithm {
-		return nil, fmt.Errorf(
-			"expected %q signing algorithm but token specified %q",
-			v.signatureAlgorithm,
-			token.Headers[0].Algorithm,
-		)
+	if err = validateSigningMethod(string(v.signatureAlgorithm), token.Headers[0].Algorithm); err != nil {
```

Comment:
Trying to refactor this func so that the steps read better.

Reply:
👏🏼